docs/spec/spec.md

```diff
@@ -0,0 +1,154 @@
+# Specification: Ballerina UUID Library
+
+_Owners_: @daneshk @MadhukaHarith92  
+_Reviewers_: @daneshk  
+_Created_: 2021/11/16  
+_Updated_: 2021/11/16  
+_Issue_: [#2351](https://github.com/ballerina-platform/ballerina-standard-library/issues/2351)
+
+# Introduction
+The UUID library is used to generate UUIDs based on the [RFC 4122](https://www.rfc-editor.org/rfc/rfc4122.html). It is part of Ballerina Standard Library. [Ballerina programming language](https://ballerina.io/) is an open-source programming language for the cloud that makes it easier to use, combine, and create network services.
+
+# Contents
+
+1. [Overview](#1-overview)
+2. [Generating UUIDs](#2-generating-uuids)
+    * 2.1. [Version 1](#21-version-1)
+    * 2.2. [Version 3](#22-version-3)
+    * 2.3. [Version 4](#23-version-4)
+    * 2.4. [Version 5](#24-version-5)
+    * 2.5. [Nil UUID](#25-nil-uuid)
+3. [Checking the Version of UUIDs](#3-checking-the-version-of-uuids)
+4. [Validating UUIDs](#4-validating-uuids)
+5. [UUID Operations](#5-uuid-operations)
+
+## 1. Overview
+This specification elaborates on the functions available in the UUID library.
+
+## 2. Generating UUIDs
+UUID module supports generating 4 versions of UUIDs. They are generated as UUID strings or UUID records. The following is the UUID record definition.
+```
+public type Uuid readonly & record {
+    ints:Unsigned32 timeLow;
+    ints:Unsigned16 timeMid;
+    ints:Unsigned16 timeHiAndVersion;
+    ints:Unsigned8 clockSeqHiAndReserved;
+    ints:Unsigned8 clockSeqLo;
+    int node;
+};
+```
+
+There are 6 components in a UUID record.
+```
+timeLow - The low field of the timestamp
+timeMid - The middle field of the timestamp
+timeHiAndVersion - The high field of the timestamp multiplexed with the version number
+clockSeqHiAndReserved - The high field of the clock sequence multiplexed with the variant
+clockSeqLo - The low field of the clock sequence
+node - The spatially unique node identifier
+```
```

Comment:
shall we add these as doc comments in the record definition?

Reply:
Fixed in bdd4890